apps/xmtp.chat/src/content-types/Intent.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  ContentTypeId,
+  type ContentCodec,
+  type EncodedContent,
+} from "@xmtp/content-type-primitives";
+import { z } from "zod";
+
+/**
+ * Intent content structure
+ * Users express their selection by sending Intent messages when they tap action buttons
+ */
+const intentSchema = z.object({
+  /** References Actions.id - provides strong coupling with Actions message */
+  id: z.string(),
+  /** References specific Action.id - indicates which action was selected */
+  actionId: z.string(),
+  /** Optional context data for the selection */
+  metadata: z
+    .record(
+      z.string(),
+      z.union([z.string(), z.number(), z.boolean(), z.null()]),
+    )
+    .optional(),
+});
+
+export type Intent = z.infer<typeof intentSchema>;
+
+/**
+ * Content Type ID for Intent messages
+ * Following XIP-67 specification for inline actions
+ */
+export const ContentTypeIntent = new ContentTypeId({
+  authorityId: "coinbase.com",
+  typeId: "intent",
+  versionMajor: 1,
+  versionMinor: 0,
+});
+
+/**
+ * Intent codec for encoding/decoding Intent messages
+ * Implements XMTP ContentCodec interface for Intent content type
+ */
+export class IntentCodec implements ContentCodec<Intent> {
+  get contentType(): ContentTypeId {
+    return ContentTypeIntent;
+  }
+
+  encode(content: Intent): EncodedContent {
+    // Validate content before encoding
+    this.#validateContent(content);
+
+    return {
+      type: ContentTypeIntent,
+      parameters: {},
+      content: new TextEncoder().encode(JSON.stringify(content)),
+    };
+  }
+
+  decode(content: EncodedContent): Intent {
+    const decodedContent = new TextDecoder().decode(content.content);
+    try {
+      const parsed = JSON.parse(decodedContent) as Intent;
+      this.#validateContent(parsed);
+      return parsed;
+    } catch (error: unknown) {
+      throw new Error(
+        `Failed to decode Intent content: ${(error as Error).message}`,
```

Comment:
We should consider an error util function. If for some reason the error is not an instance of Error, the access to `.message` will fail. Example `throw 42` and accessing `message` on it, will cause a snytax error here.

Reply:
in this case, it will only be `Error` or `SyntaxError`.

BTW, these content types were copied over from a Coinbase repo. these changes are not guaranteed to be in the final published version.